File: Python/Algorithm/Algorithm I/190. Reverse Bits.py
class Solution:
    def reverseBits(self, n: int) -> int:
        """
        Reverse the bits of a 32-bit unsigned integer.

        Args:
            n (int): A 32-bit unsigned integer.

        Returns:
            int: The integer with reversed bits.

        Time Complexity: o(1) since we are only looping 32 times.
        Space Complexity: o(1) since we are not using any additional space.
        """
        reversedValue = 0

        for _ in range(32):
            bitValue = (n & 1)
            reversedValue = (reversedValue << 1) + bitValue
            n >>= 1

        return reversedValue

File: TypeScript/Algorithm/Algorithm I/190. Reverse Bits.ts
/**
 * Reverses the bits of a given integer.
 * @param {number} n - The integer whose bits are to be reversed.
 * @returns {number} The integer with reversed bits.
 */
function reverseBits(n: number): number {
  let reverseN: number = 0;

  for (let _: number = 0; _ < 31; _++) {
    reverseN = (reverseN + (n & 1)) * 2;
    n >>= 1;
  }

  reverseN = reverseN + (n & 1);
  return reverseN;
}
